src/components/RequireAuth.tsx

```diff
@@ -18,7 +19,9 @@ export const RequireAuth = InjectAppServices(
     return appSessionStateStatus === "unknown" ? (
       <div>Loading...</div>
     ) : appSessionStateStatus !== "authenticated" ? (
-      <NavigateToExternalUrl to={loginPageUrl} />
+      <NavigateToExternalUrl
+        to={`${loginPageUrl}?redirect=${encodeURIComponent(location.href)}`}
```

Comment:
Encoding URLs seems to be working bad, I will test it a little more before merge it.

```suggestion
        to={`${loginPageUrl}?redirect=${location.href}`}
```

Reply:
Fixed!